include/verilated_vcd_c.cpp

```diff
@@ -804,6 +874,266 @@ void VerilatedVcd::fullBusX(vluint32_t code, int bits) {
 void VerilatedVcd::fullQuadX(vluint32_t code, int bits) { fullBusX(code, bits); }
 void VerilatedVcd::fullArrayX(vluint32_t code, int bits) { fullBusX(code, bits); }
 
+//=============================================================================
+// Pointer based variants used by Verilator
+
+// Emit suffix, write back write pointer, check buffer
+void VerilatedVcd::finishLine(vluint32_t* oldp, char* writep) {
+    const vluint32_t code = oldp - m_sigs_oldvalp;
+    char* const suffixp = m_suffixesp + code * 8;
+    // Copy the whole suffix (this avoid having hard to predict branches which
+    // helps a lot). Note suffixp could be aligned, so could load it in one go,
+    // but then we would be endiannes dependent which we don't have a way to
+    // test right now and probably would make little difference...
+    writep[0] = suffixp[0];
+    writep[1] = suffixp[1];
+    writep[2] = suffixp[2];
+    writep[3] = suffixp[3];
+    writep[4] = suffixp[4];
+    writep[5] = suffixp[5];
+    writep[6] = '\n';  // The 6th index is always '\n' if it's relevant, no need to fetch it.
+    // Now write back the write pointer incremented by the actual size of the suffix.
+    m_writep = writep + suffixp[7];
+    bufferCheck();
+}
+
+void VerilatedVcd::fullBit(vluint32_t* oldp, vluint32_t newval) {
+    *oldp = newval;
+    char* wp = m_writep;
+    *wp++ = '0' | static_cast<char>(newval);
```

Comment:
```suggestion
    *wp = '0' | static_cast<char>(newval);
```

Reply:
The call to `finishLine` below requires the updated pointer, as that's where it will emit the suffix.